### PR TITLE
Nerdsanddragons patch preset new features

### PR DESCRIPTION
**Pull request by nerdsanddragons**
Preset Arguments
Mapview Presets
**-newview presetname,mapview preset** - Allows you to create a preset for -mapview that is saved in your mapPresets uvar. Example: !map -newview Ambush,d5:8x8
**-deleteview presetname** - Allows you to delete an already created -mapview preset.
**-viewload presetname** - Loads a created preset for -mapview
**!map viewlist** - Allows you to see all your saved -mapview presets
​
Map BG Presets
**-newmapbg mapname:url,mapsize,mapoptions** - Allows you to create a map bg preset that is saved to your mapBackgrounds uvar to be called up using -mapload. Example - !map -newmapbg Forest:https://i.imgur.com/MlhcGVE.jpg,26x14,hd
**-deletemapbg mapname** - Allows you to delete a created map preset.

Custom Spell Overlays
**-newspell overlay/spell name:overlay** - Allows you to save custom spell overlays to your mapOverlays uvar. Example - !map -newspell Explosion:circle,30,r,
**-deletespell overlay/spell name** - Allows you to delete a created spell overlay.
Changed description to reflect new features

### CODE REVIEW

File: Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar [hunk 1]
```diff
@@ -205,6 +214,92 @@ if c:
...
+  #First option is to create the preset
+  userPresets=load_json(get("mapPresets", "{}"))
+  if args.last('newview'):
+   for x in args.get('newview'):
+    if "," in x:
+     userPresets.update({x.split(",")[0]:x.split(",")[1]})
+    else:
+     err("You must separate the key and value with a comma when adding `-newview` entries.")
+   set_uvar("mapPresets",dump_json(userPresets))
+   desc.append(f"Created the mapview preset `{x.split(',')[0]}`")
```
Comment (nerdsanddragons) [inline Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar:228]:
```
for addNewview in args.get('newview'):
    if "," in addNewview:
     userPresets.update({addNewview.split(",")[0]:addNewview.split(",")[1]})
    else:
     err("You must separate the key and value with a comma when adding `-newview` entries.")
   set_uvar("mapPresets",dump_json(userPresets))
   desc.append(f"Created the mapview preset `{addNewview.split(',')[0]}`")
```

File: Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar [hunk 2]
```diff
@@ -205,6 +214,92 @@ if c:
...
+  #Here we can remove an already created preset
+  if args.last('deleteview'):
+   for x in args.get('deleteview'):
+    userPresets.pop(x)
+   set_uvar("mapPresets",dump_json(userPresets))
+   desc.append(f"Removed Preset `{x.upper()}`")
```
Comment (nerdsanddragons) [inline Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar:234]:
```
if args.last('deleteview'):
   for searchDeleteView in args.get('deleteview'):
    userPresets.pop(searchDeleteView)
   set_uvar("mapPresets",dump_json(userPresets))
   desc.append(f"Removed Preset `{searchDeleteView.upper()}`")
```

File: Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar [hunk 3]
```diff
@@ -205,6 +214,92 @@ if c:
...
+  if a in x.lower():
+   y = userPresets.get(x)
+   break
+ maplocation = y.split(':')[0] + ":"
+ mapsize = y.split(':')[1]
+ mapSplitX, mapSplitY = mapsize.lower().split('x')
+ mapSplitX = min(int(mapSplitX) if mapSplitX.isdigit() else 10, maxSize) 
+ mapSplitY = min(int(mapSplitY) if mapSplitY.isdigit() else 10, maxSize)
+ mapsize = f"{mapSplitX}x{mapSplitY}"
+ desc.append(f"Map view changed to `{y}`")
```
Comment (nerdsanddragons) [inline Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar:252]:
```
foundViewload = "Did not find that preset"
argsViewload = args.last("viewload").lower()
for searchViewload in userPresets:
 if argsViewload in searchViewload.lower():
  foundViewload = userPresets.get(searchViewload)
  break
maplocation = foundViewload.split(':')[0] + ":"
mapsize = foundViewload.split(':')[1]
mapSplitX, mapSplitY = mapsize.lower().split('x')
mapSplitX = min(int(mapSplitX) if mapSplitX.isdigit() else 10, maxSize) 
mapSplitY = min(int(mapSplitY) if mapSplitY.isdigit() else 10, maxSize)
mapsize = f"{mapSplitX}x{mapSplitY}"
desc.append(f"Map view changed to `{foundViewload}`")
```

File: Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar [hunk 4]
```diff
@@ -205,6 +214,92 @@ if c:
...
+ #First option is to create mapbg preset
+ if args.last('newmapbg'):
+  userMaps = load_json(get('mapBackgrounds','{}'))
+  for x in args.get('newmapbg'):
+   if ":" in x:
+     userMaps.update({x.split(":")[0]:x.split(":", 1)[1].split(',')})
+   else:
+     err("You must separate the map name from the map info with a : and url, size and options with a comma when adding `-newmapbg` entries.")
+   set_uvar("mapBackgrounds",dump_json(userMaps))
+   desc.append(f"Created the mapbg `{x.split(':')[0].upper()}`")
```
Comment (nerdsanddragons) [inline Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar:266]:
```
for addMapbg in args.get('newmapbg'):
   if ":" in addMapbg:
     userMaps.update({addMapbg.split(":")[0]:addMapbg.split(":", 1)[1].split(',')})
else:
     err("You must separate the map name from the map info with a : and url, size and options with a comma when adding `-newmapbg` entries.")
   set_uvar("mapBackgrounds",dump_json(userMaps))
desc.append(f"Created the mapbg `{addMapbg.split(':')[0].upper()}`")
```

File: Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar [hunk 5]
```diff
@@ -205,6 +214,92 @@ if c:
...
+ #Second option is to delete an already created map
+ if args.last('deletemapbg'):
+  y = "Did not find that map"
+  a = args.last("deletemapbg").lower()
+  for x in userMaps:
+   if a in x.lower():
+    y = userMaps.pop(x)
+    break
+  set_uvar("mapBackgrounds",dump_json(userMaps))
+  desc.append(f"Removed Map BG `{x.upper()}`")
```
Comment (nerdsanddragons) [inline Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar:276]:
```
if args.last('deletemapbg'):
  foundDeletemapbg = "Did not find that map"
  argDeletemapbg = args.last("deletemapbg").lower()
  for searchDeletemapbg in userMaps:
   if argDeletemapbg in searchDeletemapbg.lower():
    foundDeletemapbg = userMaps.pop(searchDeletemapbg)
    break
  set_uvar("mapBackgrounds",dump_json(userMaps))
  desc.append(f"Removed Map BG `{searchDeletemapbg.upper()}`")
```

File: Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar [hunk 6]
```diff
@@ -205,6 +214,92 @@ if c:
...
+ #First option is to create spell overlay
+ if args.last('newspell'):
+  userOverlays = load_json(get('mapOverlays','{}'))
+  for x in args.get('newspell'):
+   if ":" in x:
+    userOverlays.update({x.split(":")[0]:x.split(":")[1]})
+   else:
+     err("You must separate the key and value with a colon when adding `-newspell` entries.")
+  set_uvar("mapOverlays",dump_json(userOverlays))
+  desc.append(f"Created the overlay preset `{x.split(':')[0].upper()}`")
```
Comment (nerdsanddragons) [inline Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar:290]:
```
for addNewspell in args.get('newspell'):
   if ":" in addNewspell:
    userOverlays.update({addNewspell.split(":")[0]:addNewspell.split(":")[1]})
   else:
     err("You must separate the key and value with a colon when adding `-newspell` entries.")
  set_uvar("mapOverlays",dump_json(userOverlays))
  desc.append(f"Created the overlay preset `{addNewspell.split(':')[0].upper()}`")
```

File: Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar [hunk 7]
```diff
@@ -205,6 +214,92 @@ if c:
...
+ #second option is to delete an already created spell/overlay
+ if args.last('deletespell'):
+  y = "Did not find that spell/overlay"
+  a = args.last("deletespell").lower()
+  for x in userOverlays:
+   if a in x.lower():
+    y = userOverlays.pop(x)
+    break
+  set_uvar("mapOverlays",dump_json(userOverlays))
+  desc.append(f"Removed Overlay `{x.upper()}`")
```
Comment (nerdsanddragons) [inline Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar:300]:
```
foundDeletespell = "Did not find that spell/overlay"
  argDeletespell = args.last("deletespell").lower()
  for searchDeletespell in userOverlays:
   if argDeletespell in searchDeletespell.lower():
    foundDeletespell = userOverlays.pop(searchDeletespell)
    break
  set_uvar("mapOverlays",dump_json(userOverlays))
  desc.append(f"Removed Overlay `{searchDeletespell.upper()}`")
```

File: Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar [hunk 8]
```diff
@@ -605,15 +718,21 @@ if args.get('spelllist'):
...
  mapPage = f"""-desc "{newline.join(mapPagin[ min(len(mapPagin),args.last('page', 1, int)-1)] )}" """
  # Add some pizazz
  return mapPage + f"""-title "Where are we at on the map?"
-                        -f "Add your own maps| You can create a uvar (or cvar) named `mapBackgrounds` containing a list of gvars you or other created in order to add to or change the maps here. The format would be something like `[\"a891401d-cdf2-44b2-b63b-bd86387c2659\"]`"  """
+                        -f "Add your own maps| You can add your own maps by running the command `-newmapbg <mapname>:<url>,<map size>,<map options>`"  """
+elif args.get('viewlist'):
+     # Only grab the page selected, default to first
+ presetPage = f"""-desc "{newline.join(presetPagein[ min(len(presetPagein),args.last('page', 1, int)-1)] )}" """
+ # Add some pizazz
+ return presetPage + f"""-title "Say hello to my little preset"
```
Comment (Croebh) [inline Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar:734]: Thanks for matching my shitting pun titles

**Pull request merged**